From 02a9dca05b1173bcf553a964b598683c8ceec56a Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Fri, 27 Apr 2018 17:02:40 +0300
Subject: [PATCH 146/706] enetc: sync SICAR register config with BG 133

Update SICAR registers offsets and configure
cache attributes for CBDR.

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
(cherry picked from commit 035362f2dddde94fd2842bc9438c286f3e9adce0)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc.c    | 6 +++++-
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 8 +++++---
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc.c b/drivers/net/ethernet/freescale/enetc/enetc.c
index a6924f9fabba..5b1c533406cf 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc.c
@@ -902,6 +902,10 @@ static void enetc_setup_cbdr(struct enetc_hw *hw, struct enetc_cbdr *cbdr)
 {
 	WARN_ON(lower_32_bits(cbdr->bd_dma_base) & 0x7f);
 
+	/* set CBDR cache attributes */
+	enetc_wr(hw, ENETC_SICAR2,
+		 ENETC_SICAR_RD_COHERENT | ENETC_SICAR_WR_COHERENT);
+
 	enetc_wr(hw, ENETC_SICBDRBAR0, lower_32_bits(cbdr->bd_dma_base));
 	enetc_wr(hw, ENETC_SICBDRBAR1, upper_32_bits(cbdr->bd_dma_base));
 	enetc_wr(hw, ENETC_SICBDRLENR, ENETC_RTBLENR_LEN(cbdr->bd_count));
@@ -924,7 +928,7 @@ static void enetc_configure_si(struct enetc_si *si)
 	/* set SI cache attributes */
 	enetc_wr(hw, ENETC_SICAR0,
 		 ENETC_SICAR_RD_COHERENT | ENETC_SICAR_WR_COHERENT);
-	enetc_wr(hw, ENETC_SICAR1, ENETC_SICAR_WR_MSI);
+	enetc_wr(hw, ENETC_SICAR1, ENETC_SICAR_MSI);
 	/* enable SI, start RSS by default */
 	enetc_wr(hw, ENETC_SIMR, ENETC_SIMR_EN | ENETC_SIMR_RSSE);
 }
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 8c51b4b8973a..99ec7c0b4023 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -47,8 +47,10 @@
 #define ENETC_SIMR_RSSE	BIT(0)
 
 /* Cache attribute registers for transactions initiated by ENETC */
-#define ENETC_SICAR0	0x08
-#define ENETC_SICAR1	0x0C
+#define ENETC_SICAR0	0x40
+#define ENETC_SICAR1	0x44
+#define ENETC_SICAR2	0x48
+
 /*
  * rd snoop, no alloc
  * wr snoop, no alloc, partial cache line update for BDs and full cache line
@@ -56,7 +58,7 @@
  */
 #define ENETC_SICAR_RD_COHERENT	0x2b2b0000
 #define ENETC_SICAR_WR_COHERENT	0x00006727
-#define ENETC_SICAR_WR_MSI	0x00000037 /* wr no snoop, no alloc */
+#define ENETC_SICAR_MSI	0x00300030 /* rd/wr device, no snoop, no alloc */
 
 #define ENETC_SIPMAR0	0x80
 #define ENETC_SIPMAR1	0x84
-- 
2.17.1

